From f0dd7cde32f760b0e40c933eee0d59846303ce0f Mon Sep 17 00:00:00 2001
From: Marcel Bargull <marcel.bargull@udo.edu>
Date: Wed, 22 Jul 2020 18:53:39 +0200
Subject: [PATCH] Increase buffer size for readlink to 1024

If Nim is installed in a long prefix on Linux, "nim c" was not able to
read its nim.cfg since it tried to read the link target of
/proc/self/exe which may be longer than the previous 256 long buffer.
---
 lib/pure/os.nim | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/lib/pure/os.nim b/lib/pure/os.nim
index 8bbceddebdc..59da89c25c9 100644
--- a/lib/pure/os.nim
+++ b/lib/pure/os.nim
@@ -1213,11 +1213,11 @@ proc findExe*(exe: string, followSymlinks: bool = true;
         when not defined(windows):
           while followSymlinks: # doubles as if here
             if x.checkSymlink:
-              var r = newString(256)
-              var len = readlink(x, r, 256)
+              var r = newString(1024)
+              var len = readlink(x, r, 1024)
               if len < 0:
                 raiseOSError(osLastError(), exe)
-              if len > 256:
+              if len > 1024:
                 r = newString(len+1)
                 len = readlink(x, r, len)
               setLen(r, len)
@@ -2555,11 +2555,11 @@ proc expandSymlink*(symlinkPath: string): string {.noWeirdTarget.} =
   when defined(windows):
     result = symlinkPath
   else:
-    result = newString(256)
-    var len = readlink(symlinkPath, result, 256)
+    result = newString(1024)
+    var len = readlink(symlinkPath, result, 1024)
     if len < 0:
       raiseOSError(osLastError(), symlinkPath)
-    if len > 256:
+    if len > 1024:
       result = newString(len+1)
       len = readlink(symlinkPath, result, len)
     setLen(result, len)
@@ -2850,9 +2850,9 @@ when not weirdTarget and (defined(freebsd) or defined(dragonfly)):
 
 when not weirdTarget and (defined(linux) or defined(solaris) or defined(bsd) or defined(aix)):
   proc getApplAux(procPath: string): string =
-    result = newString(256)
-    var len = readlink(procPath, result, 256)
-    if len > 256:
+    result = newString(1024)
+    var len = readlink(procPath, result, 1024)
+    if len > 1024:
       result = newString(len+1)
       len = readlink(procPath, result, len)
     setLen(result, len)
